src/screens/FeedScreen/Header.js

```diff
@@ -174,12 +175,15 @@ const _renderAnonimity = ({
                 <GlobalButton
                   testID="onBack"
                   onPress={() => {
-                    navigation.navigate('HomeTabs', {
-                      screen: 'Feed',
-                      params: {
-                        isGoBack: true
-                      }
-                    });
+                    if (isFromFeeds) {
+                      return navigation.navigate('HomeTabs', {
+                        screen: 'Feed',
+                        params: {
+                          isGoBack: true
+                        }
+                      });
+                    }
+                    return navigation.goBack();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The navigation logic has been modified to conditionally navigate back to the 'Feed' screen if `isFromFeeds` is true, otherwise it will simply go back. This seems like a reasonable change, but it's important to ensure that this new behavior doesn't conflict with any existing navigation flows in your application.

```diff
-                    navigation.navigate('HomeTabs', {
-                      screen: 'Feed',
-                      params: {
-                        isGoBack: true
-                      }
-                    });
+                    if (isFromFeeds) {
+                      return navigation.navigate('HomeTabs', {
+                        screen: 'Feed',
+                        params: {
+                          isGoBack: true
+                        }
+                      });
+                    }
+                    return navigation.goBack();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->